### PR TITLE
Add arrow_options to read_gbq

### LINKED ISSUE #45: Support configuring `types_mapper` in `read_gbq`
https://github.com/googleapis/python-bigquery/pull/1529 and https://github.com/googleapis/python-bigquery/pull/1547 have recently added arguments for overriding the default type conversions performed by `record_batch.to_pandas()`; this allows, for example, loading string data directly into dtype `string[pyarrow]` (which can be quite a bit more [efficient](https://www.coiled.io/blog/pyarrow-strings-in-dask-dataframes)) without doing any expensive conversions after the fact.

I think we could basically just copy the implementation from the above PRs, same kwarg names and everything. Anyone see any potential issues @jrbourbeau @j-bennet @ncclementi?

### PR DESCRIPTION
**Pull request by bnaul**
Fixes #45.

Followed the same pattern pointed out by @j-bennet [here](https://github.com/coiled/dask-snowflake/blob/42cb99e4e35aeb18f7ede95badba8f20a4113378/dask_snowflake/core.py#L215-L217).

### CODE REVIEW

**Comment by bnaul** [comment]
Oops let me re-push to my fork and re-open

**Comment by jrbourbeau** [comment]
I think repo secrets are used here in CI, so PRing from a branch and not a fork is the right way to go 👍

**Comment by bnaul** [comment]
oh heh. well neither one is working right now, I thought maybe the error I was seeing was bc I didn't have the right permissions to create CI jobs on this repo but it seems to be something else....

**Comment by bnaul** [comment]
@j-bennet @jrbourbeau CI is happy now post-#48 if you would like to take a look 🙂

File: dask_bigquery/core.py [hunk 1]
```diff
@@ -66,6 +67,8 @@ def bigquery_read(
       Name of the BigQuery project.
     read_kwargs: dict
       kwargs to pass to read_rows()
+    arrow_options: dict
+      kwargs to pass to record_batch.to_pandas() when converting from pyarrow to pandas
```
Comment (j-bennet) [inline dask_bigquery/core.py:71]: Maybe add a link to https://arrow.apache.org/docs/python/generated/pyarrow.RecordBatch.html#pyarrow.RecordBatch.to_pandas, so people can easily look up those kwargs.

**Review by j-bennet — APPROVED** [review]
LGTM, thank you @bnaul.

**Pull request merged**